src/functions/asyncCredential/sessionService/sessionService.ts

```diff
@@ -64,6 +66,49 @@ export class SessionService implements IRecoverAuthSession {
     return successResponse(result.Items[0].sessionId.S);
   }
 
+  async createSession(
+    sessionConfig: ISessionConfig,
+  ): Promise<ErrorOrSuccess<null>> {
+    const config = {
+      TableName: this.tableName,
+      Item: {
+        authSessionId: { S: sessionConfig.authSessionId },
+        state: { S: sessionConfig.state },
+        sub: { S: sessionConfig.sub },
+        client_id: { S: sessionConfig.client_id },
+        govuk_signin_journey_id: { S: sessionConfig.govuk_signin_journey_id },
+        redirect_uri: { S: sessionConfig.redirect_uri },
+        issuer: { S: sessionConfig.issuer },
+        sessionState: { S: sessionConfig.sessionState },
+      },
+    };
+
+    let doesSessionExist;
+    try {
+      doesSessionExist = await this.checkSessionsExists(
+        sessionConfig.authSessionId,
+      );
+    } catch (error) {
+      return errorResponse(
+        "Unexpected error when querying session table to check if authSessionId exists",
+      );
+    }
+
+    if (doesSessionExist) {
+      return errorResponse("authSessionId already exists in the database");
+    }
+
+    try {
+      await dbClient.send(new PutItemCommand(config));
```

Comment:
please may we move this into a private function

Reply:
Done